frontend/server/src/main/java/org/pytorch/serve/metrics/Metric.java

```diff
@@ -52,6 +55,16 @@ public Metric(
                 String.valueOf(TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()));
     }
 
+    private Metric(telemetryMetricsBuilder builder) {
+        this.metricName = builder.metricName;
+        this.value = builder.value;
+        this.unit = builder.unit;
+        this.hostName = builder.hostName;
+        this.dimensions = builder.dimensions;
+        this.timestamp =
+                String.valueOf(TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()));
+    }
+
```

Comment:
It is not necessary. Telemetry metric is just a metric which has no any difference with Metric. Existing Metric is able to support it completely.